abel/rbasex.py

```diff
@@ -0,0 +1,751 @@
+        shape. Parts of the image can be excluded from analysis by assigning
+        zero weights to their pixels.
+    direction : str: ``'forward'`` or ``'inverse'``
+        type of Abel transform to be performed
+    reg : None or str or tuple (str, float), optional
+        regularization to use for inverse Abel transform. ``None`` means no
+        regularization, a string selects a non-parameterized regularization
+        method, and parameterized methods are selected by a tuple (`method`,
+        `strength`). Available methods are:
+
+        ``'L2'``:
+            Tikhonov :math:`L_2` regularization with `strength` as the square
+            of the Tikhonov factor. This is the same as “Tikhonov
+            regularization” used in BASEX, with almost identical effects on the
+            radial distributions.
+        ``'diff'``:
+            Tikhonov regularization with the difference operator (approximation
+            of the derivative) multiplied by the square root of `strength` as
+            the Tikhonov matrix. This tends to produce less blurring, but more
+            negative overshoots than ``'L2'``.
+        ``'SVD'``:
+            truncated SVD (singular value decomposition) with
+            N = `strength` × **rmax** largest singular values removed for each
+            angular order. This mimics the approach proposed in pBasex.
+        ``'pos'``:
+            non-parameterized method, finds the best (in the least-squares
+            sense) solution with non-negative :math:`\cos^n\theta \sin^m\theta`
+            terms (see :meth:`~abel.tools.vmi.Distributions.Results.cossin`).
+            For **order** = 0, 1, and 2 (with **odd** = `False`) this is
+            equivalent to :math:`I(r, \theta) \geqslant 0`; for higher orders
+            this assumption is stronger than :math:`I \geqslant 0` and
+            corresponds to no interference between different multiphoton
+            channels. Not implemented for odd orders > 1.
+
+            Notice that this method is nonlinear, which also means that it is
+            considerably slower than the linear methods and the transform
+            operator cannot be cached.
+
+        In all cases, `strength` = 0 provides no regularization. For the
+        Tikhonov methods, `strength` ~ 100 is a reasonable value for megapixel
+        images. For truncated SVD, `strength` must be < 1; `strength` ~ 0.1 is
+        a reasonable value; `strength` ~ 0.5 can produce noticeable ringing
+        artifacts.
+    out : str or None
+        shape of the output image:
+
+        ``'same'`` (default):
+            same shape and origin as the input
+        ``'fold'`` (fastest):
+            Q0 quadrant (for ``odd=False``) or right half (for ``odd=True``) up
+            to **rmax**, but limited to the largest input-image quadrant (or
+            half)
+        ``'unfold'``:
+            like ``'fold'``, but symmetrically “unfolded”
```

Comment:
I'm not really sure what this means. Basically, you calculate just Q0 and then duplicate it to all of the other quadrants? Isn't this what the transform always does? Or does it normally allow for asymmetry in the top/down and left/right dimensions?

Reply:
Like lin-BASEX, it assumes that the image is composed of spherical harmonics, so for even-only orders it is totally symmetric, for even-only orders is horizontally symmetric and vertically antisymmetric, and for mixed orders is just horizontally symmetric. In principle, you can transform each quadrant independently, but this will yield 4 different velocity distributions. Averaging them will give the same results as transforming the whole image, so this is not very useful.
The transform itself actually produces only the velocity distribution. The transformed image is then created from that distribution in the form requested (internally, yes, just Q0 of right half is calculated and, if needed, is mirrored to other parts).